src/TableV8/styled.tsx

```diff
@@ -116,25 +153,108 @@ const TableRow = (
     withStripes?: boolean;
   }>,
 ) => <tr {...props} className={clsx(TABLE_CLASSES.TR, props.className)} />;
-export const StyledTableRow = styled(TableRow, {
-  shouldForwardProp: (prop) =>
-    isPropValid(prop) &&
-    !['index', 'withRowBorder', 'withRowHighlight', 'withStripes'].includes(prop),
+export const TableRow = styled(TableRowComp, {
+  shouldForwardProp: (prop) => isPropValid(prop),
 })`
-  display: flex;
   background: ${(props) =>
-    props.index % 2 === 0 && props.withStripes ? COLORS.BACKGROUND : 'none'};
-  border-bottom: ${(props) => `1px solid ${props.withRowBorder ? COLORS.BORDER : 'transparent'}`};
-  &:hover {
-    background: ${(props) =>
-      props.withRowHighlight ? COLORS.ROW_HIGHLIGHT_BACKGROUND : 'initial'};
-  }
+    props.index % 2 && props.withStripes ? COLORS.BACKGROUND_ALT : 'transparent'};
+  border-bottom: ${(props) => (props.withRowBorder ? `1px solid ${COLORS.BORDER}` : '0 none')};
+  ${(props) =>
+    props.withRowHighlight &&
+    `
+    &:hover {
+      background: ${COLORS.BACKGROUND_HIGHLIGHT}
+    }
+  `}
 `;
 
-const TableCell = (props: React.PropsWithChildren<{ className?: string }>) => (
+// the TD element should only for table structure.
+// use TableCellWrapper for additional styling within the TD.
+const TableCellComp = (props: React.PropsWithChildren<{ className?: string; width?: number }>) => (
   <td {...props} className={clsx(TABLE_CLASSES.TD, props.className)} />
 );
-export const StyledTableCell = styled(TableCell)`
-  white-space: break-spaces;
-  word-break: break-word;
+export const TableCell = styled(TableCellComp, {
+  shouldForwardProp: (prop) => isPropValid(prop) && !['width'].includes(prop),
+})`
+  width: ${(props) => `${props.width || 1}px`};
+`;
+
+// import this to add custom styles to a table cell
+const TableCellWrapperComp = (props: React.PropsWithChildren<{ className?: string }>) => (
+  <div {...props} className={clsx(TABLE_CLASSES.TD_WRAPPER, props.className)} />
+);
+export const TableCellWrapper = styled(TableCellWrapperComp, {
+  shouldForwardProp: (prop) => isPropValid(prop),
+})``;
+
+const ResizerComp = (
+  props: React.PropsWithChildren<{
+    className?: string;
+    onMouseDown: (e: unknown) => void;
+    onTouchStart: (e: unknown) => void;
+  }>,
+) => <button {...props} className={clsx(TABLE_CLASSES.RESIZER, props.className)} type="button" />;
+export const Resizer = styled(ResizerComp)`
+  border: 0 none;
+  position: absolute;
+  right: -6px;
+  top: 0;
+  height: 100%;
+  width: 13px;
+  cursor: col-resize;
+  background: transparent;
+  user-select: none; // send event to <ResizerComp />
+  touch-action: none; // send event to <ResizerComp />
+  z-index: 10;
+`;
+
+const LoaderComp = (props: React.PropsWithChildren<{ className?: string; loading?: boolean }>) => (
+  <div {...props} className={clsx(TABLE_CLASSES.TABLE_CONTAINER, props.className)}>
```

Comment:
i think you'll end up with 
```
<div className='table-container'><div className='table-container'><Loader></div></div>
```
The TABLE_CLASSES don't interact at all with styling , is that correct?

Reply:
whoops, I want that to be TABLE_CLASSES.LOADER or similar.

the table classes do interact with styling, I use them instead of component selectors since those don't work in our project.